src/frontend/components/StudentLiveWrapper/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect } from 'react';
+
+import DashboardVideoLiveJitsi from 'components/DashboardVideoLiveJitsi';
+import { LiveVideoPanel } from 'components/LiveVideoPanel';
+import { StudentLiveControlBar } from 'components/StudentLiveControlBar';
+import { StudentLiveInfoBar } from 'components/StudentLiveInfoBar';
+import VideoPlayer from 'components/VideoPlayer';
+import {
+  LivePanelItem,
+  useLivePanelState,
+} from 'data/stores/useLivePanelState';
+import { useVideo } from 'data/stores/useVideo';
+import { Video } from 'types/tracks';
+
+import { StudentLiveLayout } from './StudentLiveLayout';
+
+interface LiveStreamerConfiguration {
+  type: 'on_stage';
+}
+
+interface LiveViewerConfiguration {
+  type: 'viewer';
+  playerType: string;
+}
+
+interface LiveVideoWrapperProps {
+  video: Video;
+  configuration: LiveStreamerConfiguration | LiveViewerConfiguration;
+}
+
+export const LiveVideoWrapper: React.FC<LiveVideoWrapperProps> = ({
+  video: baseVideo,
+  configuration,
+}) => {
+  const video = useVideo((state) => state.getVideo(baseVideo));
+  const { isPanelVisible, setPanelVisibility, configPanel } = useLivePanelState(
+    (state) => ({
+      isPanelVisible: state.isPanelVisible,
+      setPanelVisibility: state.setPanelVisibility,
+      configPanel: state.setAvailableItems,
+    }),
+  );
+
+  useEffect(() => {
+    const availableItems: LivePanelItem[] = [];
+    let currentItem;
+    if (video.xmpp !== null) {
+      availableItems.push(LivePanelItem.CHAT);
+      //  if available, LivePanelItem.CHAT needs to be selected by default
+      //  because it will be rendered hidden and connection to XMPP will be initialized
+      currentItem = LivePanelItem.CHAT;
+    }
+    configPanel(availableItems, currentItem);
+  }, [video, setPanelVisibility, configPanel]);
```

Comment:
```suggestion
  }, [video]);
```

`setPanelVisibility` and `configPanel` is not a value.

Reply:
I'm not sure to understand, yes both are a function but yet they are capture in the closure.

You can argue that in the current implementation of zustand (and our store in particular) those functions are declared when the store is created and never updated, and you would be wright.

But, on the other hand, it makes no difference (since they will never changed, they will never trigger the useEffect), and I find it safer to keep all required dependencies in a useEffect

Reply:
Yes their references will never change but it's ok you can keep them. But not `setPanelVisibility`. it is never used.